numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -1792,8 +1792,12 @@ type_tuple_userloop_type_resolver(PyUFuncObject *self,
             if (obj == NULL) {
                 continue;
             }
-            funcdata = (PyUFunc_Loop1d *)NpyCapsule_AsVoidPtr(obj);
-            while (funcdata != NULL) {
+
+            for (
+                funcdata = (PyUFunc_Loop1d *)NpyCapsule_AsVoidPtr(obj);
+                funcdata != NULL;
+                funcdata = funcdata->next
+            ) {
```

Comment:
The `)` should probably go on the preceding line. Maybe something like
```
            for (funcdata = (PyUFunc_Loop1d *)NpyCapsule_AsVoidPtr(obj);
                 funcdata != NULL;
                 funcdata = funcdata->next) {
```
If the initialization is put on the previous line, the rest will fit on one.